tools/plugins/preinstall/src/main/kotlin/net/corda/cli/plugins/preinstall/CheckKafka.kt

```
@@ -0,0 +1,151 @@
+
+    @Option(names = ["-v", "--verbose"], description = ["Display additional information when checking resources"])
+    var verbose: Boolean = false
+
+    @Option(names = ["-d", "--debug"], description = ["Show information about kafka config creation for debugging purposes"])
+    var debug: Boolean = false
+
+    private fun getKafkaProperties(yaml: Kafka, saslUsername: String, saslPassword: String, truststorePassword: String): Properties? {
+        val props = Properties()
+        props["bootstrap.servers"] = yaml.kafka.bootstrapServers
+        props["request.timeout.ms"] = timeout
+        props["connections.max.idle.ms"] = 5000
+
+        // Assembles the properties in the same way as the helm charts:
+        // https://github.com/corda/corda-runtime-os/blob/release/os/5.0/charts/corda-lib/templates/_bootstrap.tpl#L333
+        if (yaml.kafka.tls.enabled) {
+            if (yaml.kafka.sasl.enabled) {
+                props["security.protocol"] = "SASL_SSL"
+                props["sasl.mechanism"] = yaml.kafka.sasl.mechanism
+                if (yaml.kafka.sasl.mechanism == "PLAIN") {
+                    props["sasl.jaas.config"] =
+                        "org.apache.kafka.common.security.plain.PlainLoginModule required username=" +
+                                "\"$saslUsername\" password=\"$saslPassword\" ;"
+                } else {
+                    props["sasl.jaas.config"] =
+                        "org.apache.kafka.common.security.plain.ScramLoginModule required username=" +
+                                "\"$saslUsername\" password=\"$saslPassword\" ;"
+                }
+            } else {
+                props["security.protocol"] = "SSL"
+            }
+            yaml.kafka.tls.truststore?.valueFrom?.secretKeyRef?.name?.let {
+                props["ssl.truststore.location"] = truststoreLocation ?: run {
+                    log("If SSL is enabled, you must provide the location of the truststore file with -f.", ERROR)
```

it would be better to propagate an exception here than have to handle the nulls